### PR TITLE
fix: add support for login via username

### PR DESCRIPTION
**Pull request by mumer92**
This PR adds support for username and email to be used to login to app.

- User can login using username
- User can login using email

Simulator Screenshot - iPhone 15 Pro 17 - 2023-11-01 at 18 00 46

### CODE REVIEW

**Review by saeedbashir — CHANGES_REQUESTED** [review]

File: Authorization/Authorization/Presentation/Login/SignInViewModel.swift [hunk 1]
```diff
@@ -50,12 +50,12 @@ public class SignInViewModel: ObservableObject {
     
     @MainActor
     func login(username: String, password: String) async {
-        guard validator.isValidEmail(username) else {
-            errorMessage = AuthLocalization.Error.invalidEmailAddress
+        guard !username.isEmpty else {
```
[the remark this answers is not in the document]
Comment (saeedbashir) [reply Authorization/Authorization/Presentation/Login/SignInViewModel.swift:53]: @mumer92 Can you look into this as well.

**Comment by saeedbashir** [comment]
@mumer92 It's not giving an error for an empty `username`.

**Review by saeedbashir — APPROVED** [review]
LGTM 👍

**Comment by mumer92** [comment]
@volodymyr-chekyrta waiting for your review on this.

File: Core/Core/View/Validator.swift [hunk 2]
```diff
@@ -9,24 +9,23 @@ import Foundation
 
 public class Validator {
     
-    private let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+    private let emailRegEx = ".+@.+\\..+"
```
Comment (volodymyr-chekyrta) [inline Core/Core/View/Validator.swift:12]: What is the reason for changing this regex expression?

Comment (mumer92) [reply Core/Core/View/Validator.swift]: reverted it back to original.

File: Authorization/Authorization/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -9,14 +9,15 @@
 "SIGN_IN.LOG_IN_TITLE" = "Sign in";
 "SIGN_IN.WELCOME_BACK" = "Welcome back! Please authorize to continue.";
 "SIGN_IN.EMAIL" = "Email";
```
Comment (volodymyr-chekyrta) [inline Authorization/Authorization/en.lproj/Localizable.strings:11]: Looks like this string is no longer in use and should be removed.

**Pull request merged**